go/private/sdk.bzl

```diff
@@ -441,7 +441,10 @@ def _remote_sdk(ctx, urls, strip_prefix, sha256):
     # in Bazel 6.0.0+ (bazelbuild/bazel#16052). The only situation where
     # .zip files are needed seems to be a macOS host using a Windows toolchain
     # for remote execution.
-    if urls[0].endswith(".tar.gz"):
+    if (urls[0].endswith(".tar.gz") and
+          # Development versions of Bazel have an empty version string. We assume that they are
+          # more recent than the version that introduced the fix.
+          not versions.is_at_least("6.4.0", versions.get() or "6.4.0")):
```

Comment:
Thanks for sending this, I completely forgot to clean this up after the fix.

We aim to consolidate all feature detection in https://github.com/bazel-contrib/bazel_features. Could you submit a PR to that repo adding a new feature (e.g. `extract_supports_unicode_filenames`) here:
https://github.com/bazel-contrib/bazel_features/blob/697d54eaf05ed83972204f8deee9817dcd6edfde/features.bzl#L21

We can then bump the `bazel_features` dependency and use the named feature.

Reply:
In fact, if we do this, we can also get rid of the `zip` workaround in this case.

Reply:
bazel-contrib/bazel_features#41

Reply:
> In fact, if we do this, we can also get rid of the zip workaround in this case.
Sorry, I don't understand. I was going to change the logic to:
```
if bazel_features.external_deps.extract_supports_unicode_filenames:
  # Do `download_and_extract`
elif urls[0].endswith(".tar.gz"):
  # Do `tar` workaround
elif urls[0].endswith(".zip") and host_goos != "windows":
  # Do `zip` workaround
```
What should I change it too. I'm happy for you to pick this up if it would be more expedient.

Reply:
Yeah, that logic is exactly what I had in mind!

The feature is available in bazel_features v1.6.0 now.

Reply:
```
ERROR: Failed to load Starlark extension '@io_bazel_rules_go_bazel_features//:features.bzl'.

Cycle in the workspace file detected. This indicates that a repository is used prior to being defined.
The following chain of repository dependencies lead to the missing definition.
 - @io_bazel_rules_go_bazel_features
This could either mean you have to add the '@io_bazel_rules_go_bazel_features' repository with a statement like `http_archive` in your WORKSPACE file (note that transitive dependencies are not added automatically), or move an existing definition earlier in your WORKSPACE file.
```

Is this anything to do with the double-step setup polyfilling we're doing in the `WORKSPACE`.

I'm a bit in the dark what's going on here. Any tips?

Reply:
Ah, yes, I think it's because the call that registers our deps ends up loading the same file and that leads to the cycle. I will take a look. If that doesn't work, we may have to revert back to the hardcoded version check. Sorry for the trouble, WORKSPACE is painful.

Reply:
I've push a commit on top that refactors back the `versions` check. Feel free to remove that commit, revert or whatever else if you can get `bazel_features` to work!

Reply:
Had to add a final statement to logic statement to pick up Windows.

Reply:
There is a chance with a change in logic that a setup that used to fall through to `download_and_extract` now will fail: that setup just might not be in BuildKite.

Reply:
The Bazel 5 pipeline is failing. If we can also make that work, this is definitely good to go. If not, we can still consider dropping support for Bazel 5.

Reply:
Seems like I caught all the edge cases.